relay-server/src/services/processor.rs

```diff
@@ -101,38 +103,82 @@ macro_rules! if_processing {
 /// The minimum clock drift for correction to apply.
 const MINIMUM_CLOCK_DRIFT: Duration = Duration::from_secs(55 * 60);
 
+/// Transaction group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct TransactionGroup;
+
+/// Error group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct ErrorGroup;
+
+/// Session group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct SessionGroup;
+
+/// Standalone group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct StandaloneGroup;
+
+/// ClientReport group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct ClientReportGroup;
+
+/// Replay group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct ReplayGroup;
+
+/// CheckIn group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct CheckInGroup;
+
+/// Span group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct SpanGroup;
+
+/// Metrics group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct MetricsGroup;
+
+/// Unknown group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct ForwardUnknownGroup;
+
+/// Ungrouped group type marker.
+#[derive(Clone, Copy, Debug)]
+pub struct Ungrouped;
+
 /// Describes the groups of the processable items.
 #[derive(Clone, Copy, Debug)]
 pub enum ProcessingGroup {
     /// All the transaction related items.
     ///
     /// Includes transactions, related attachments, profiles.
-    Transaction,
+    Transaction(TransactionGroup),
```

Comment:
I don't think this does anything for us now, do you just want to have it here for the future?

Reply:
Yeah, it's the preparation for the followups